File: UIAutomation/Cucumber/src/test/java/Case/step_def/ProductSteps.java
package Case.step_def;

import Case.pages.BasketPage;
import Case.pages.CategoryPage;
import Case.pages.HomePage;
import Case.pages.ProductDetailPage;
import Case.utils.Helper;
import io.cucumber.java.en.And;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class ProductSteps {
    HomePage _homePage;
    CategoryPage _categoryPage;
    ProductDetailPage _productDetailPage;
    BasketPage _basketPage;
    Helper _helper;

    public ProductSteps() {
        _homePage = new HomePage();
        _categoryPage = new CategoryPage();
        _productDetailPage = new ProductDetailPage();
        _basketPage = new BasketPage();
        _helper = new Helper();
    }

    @Given("{string} adresine gidilir.")
    public void adresineGidilir(String url) {
        _helper.navigateToUrl(url);
        _homePage.tumCerezlereIzinVer();
        _homePage.anaSayfaAcildiMi(); //assert
    }

    @And("Herhangi bir kategori sayfasına gidilir.")
    public void herhangiBirKategoriSayfasinaGidilir() {
        _homePage.herhangiBirKategoriMenuSecim();
        _homePage.herhangiBirKategoriSecim();
        _categoryPage.kategoriAcildiMi(); //assert
    }

    @And("Herhangi bir ürün sayfasına gidilir.")
    public void herhangiBirUrunSayfasinaGidilir() {
        String urunBaslik = _categoryPage.herhangiBirUrunDetaySayfasinaGit();
        _productDetailPage.baslikKontrol(urunBaslik); //assert
    }

    @And("Ürün sepete eklenir.")
    public void urunSepeteEklenir() {
        _productDetailPage.sepeteEkleButonunaTikla();
        _productDetailPage.sepeteEklendiMi(); //assert
    }

    @When("Sepet sayfasına gidilir.")
    public void sepetSayfasinaGidilir() {
        _productDetailPage.sepeteGit();
        _basketPage.sepeteGidildiMi(); //assert
    }

    @Then("Anasayfaya geri dönülür.")
    public void anasayfayaGeriDonulur(){
        _homePage.anaSayfayaGit();
        _homePage.anaSayfaAcildiMi(); //assert
    }
}

File: UIAutomation/Behave/steps/ProductSteps.py
from behave import *

from pages.CategoryPage import CategoryPage
from pages.HomePage import HomePage
from pages.BasketPage import BasketPage
from pages.ProductDetailPage import ProductDetailPage
from utils.Helper import Helper


class ProductSteps(HomePage):

    @given('"{url}" adresine gidilir.')
    def step_impl(self, url):
        Helper.navigate_to_url(url)
        HomePage.tum_cerezlere_izin_ver(self)
        HomePage.ana_sayfa_acildi_mi(self)

    @step("Herhangi bir kategori sayfasına gidilir.")
    def step_impl(self):
        HomePage.herhangi_bir_kategori_menu_secim(self)
        HomePage.herhangi_bir_kategori_secim(self)
        CategoryPage.kategori_acildi_mi(self)

    @step("Herhangi bir ürün sayfasına gidilir.")
    def step_impl(self):
        urunBaslik = CategoryPage.herhangi_bir_urun_detay_sayfasina_git(self)
        ProductDetailPage.baslik_kontrol(self, urunBaslik)

    @step("Ürün sepete eklenir.")
    def step_impl(self):
        ProductDetailPage.sepete_ekle_butonuna_tikla(self)
        ProductDetailPage.sepete_eklendi_mi(self)

    @when("Sepet sayfasına gidilir.")
    def step_impl(self):
        ProductDetailPage.sepete_git(self)
        BasketPage.sepete_gidildi_mi(self)

    @then("Anasayfaya geri dönülür.")
    def step_impl(self):
        HomePage.ana_sayfaya_git(self)
        HomePage.ana_sayfa_acildi_mi(self)
